spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/WebMvcAutoConfiguration.java

```diff
@@ -506,16 +510,31 @@ private ResourceResolver getVersionResourceResolver(
 		private static final Log logger = LogFactory
 				.getLog(WelcomePageHandlerMapping.class);
 
-		private WelcomePageHandlerMapping(Resource welcomePage) {
-			if (welcomePage != null) {
+		private WelcomePageHandlerMapping(final ApplicationContext applicationContext,
+				final Resource welcomePage) {
+			if (welcomeTemplateExists(applicationContext)) {
```

Comment:
I wouldn't worry about that for now, certainly not in the context of this change anyway